packages/block-editor/src/hooks/layout-child.js

```diff
@@ -37,13 +39,14 @@ function useBlockPropsChildLayoutStyles( { style } ) {
 			}`;
 		}
 		/**
-		 * If parentColumnWidth is set, the grid is responsive
-		 * so a container query is needed for the span to resize.
+		 * If minimumColumnWidth is set on the parent, or if no
+		 * columnCount is set, the grid is responsive so a
+		 * container query is needed for the span to resize.
 		 */
-		if ( columnSpan && parentColumnWidth ) {
+		if ( columnSpan && ( minimumColumnWidth || ! columnCount ) ) {
```

Comment:
Similar to the PHP side of things, do we also need to check that the parent layout type is `grid`?

Reply:
We could actually do this here because the parentLayout comes from the BlockList context and seems to always have the type, but not sure if it's worth having different logic here and in the PHP.

Reply:
Yeah, I wouldn't worry about it for now. We can always follow up if need be!